shared/src/main/java/net/skinsrestorer/shared/plugin/SRServerPlugin.java

```diff
@@ -62,16 +64,38 @@ public void checkProxyMode() {
         proxyMode = checkProxy();
 
         try {
-            Path warning = plugin.getDataFolder().resolve("(README) Use proxy config for settings! (README).txt");
+            final Path warning = plugin.getDataFolder().resolve("(README) Use proxy config for settings! (README).txt");
             if (proxyMode) {
-                Files.createDirectories(plugin.getDataFolder());
+                if (!Files.isDirectory(plugin.getDataFolder())) { // in case the directory is a symbol link
+                    Files.createDirectories(plugin.getDataFolder());
+                }
 
-                try (InputStream in = serverAdapter.getResource("proxy_warning.txt")) {
+                try (final InputStream in = serverAdapter.getResource("proxy_warning.txt")) {
                     if (in == null) {
                         throw new IllegalStateException("Could not find proxy_warning.txt in resources!");
                     }
-                    // Always replace the file to make sure it's up-to-date.
-                    Files.copy(in, warning, StandardCopyOption.REPLACE_EXISTING);
+
+                    final byte[] proxyWarningBuffer = new byte[8192];
+                    final int nBytes = in.read(proxyWarningBuffer);
+                    if (Files.exists(warning)) {
+                        final boolean needReplace;
+                        try (final InputStream existingWarningStream = Files.newInputStream(warning)) {
+                            needReplace = !IntStream.range(0, nBytes).map(i -> proxyWarningBuffer[i]).allMatch(origin -> {
+                                try {
+                                    return origin == existingWarningStream.read();
+                                } catch (IOException e) {
+                                    throw new RuntimeException(e);
+                                }
+                            });
+                        }
+                        if (needReplace) {
+                            try (final OutputStream fos = Files.newOutputStream(warning)) {
+                                fos.write(proxyWarningBuffer, 0, nBytes);
+                            }
+                        }
```

Comment:
Could you remake this part to just read both files as a string and then check whether they match with #equals()? I know this is probably at worst one O(n) slower, but it will look a lot clearer to others than comparing bytes. Also can you just call Files.copy() as well there instead of opening a OutputStream?

Reply:
Since Files#readString is not available in Java 8 and #getResource returns an Input stream, I think that reading them to strings will not be clearer.

Reply:
And if I just call Files.copy(), I will need to open a new `InputStream`.

Reply:
new String(Files.readAllBytes(), StandardCharSets.UTF8) iirc

Reply:
Done.